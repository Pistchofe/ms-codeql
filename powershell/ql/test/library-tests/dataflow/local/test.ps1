$a1 = Source
Sink $a1

$b = GetBool
if($b) {
    $a2 = Source
}
Sink $a2