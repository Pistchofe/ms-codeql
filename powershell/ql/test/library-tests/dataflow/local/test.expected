| test.ps1:1:1:1:4 | a1 | test.ps1:2:6:2:9 | a1 |
| test.ps1:1:7:1:13 | Source | test.ps1:1:1:1:4 | a1 |
| test.ps1:1:7:1:13 | Source | test.ps1:1:1:1:13 | ...=... |
| test.ps1:2:1:2:9 | Sink | test.ps1:2:1:2:9 | pre-return value for Sink |
| test.ps1:2:1:2:9 | Sink | test.ps1:2:1:2:9 | pre-return value for Sink |
| test.ps1:2:1:2:9 | implicit unwrapping of Sink | test.ps1:1:1:11:8 | return value for test.ps1 |
| test.ps1:2:1:2:9 | pre-return value for Sink | test.ps1:2:1:2:9 | implicit unwrapping of Sink |
| test.ps1:4:1:4:3 | b | test.ps1:5:4:5:6 | b |
| test.ps1:4:6:4:13 | GetBool | test.ps1:4:1:4:3 | b |
| test.ps1:4:6:4:13 | GetBool | test.ps1:4:1:4:13 | ...=... |
| test.ps1:5:4:5:6 | b | test.ps1:10:14:10:16 | b |
| test.ps1:6:5:6:8 | a2 | test.ps1:8:6:8:9 | a2 |
| test.ps1:6:11:6:17 | Source | test.ps1:6:5:6:8 | a2 |
| test.ps1:6:11:6:17 | Source | test.ps1:6:5:6:17 | ...=... |
| test.ps1:8:1:8:9 | Sink | test.ps1:8:1:8:9 | pre-return value for Sink |
| test.ps1:8:1:8:9 | Sink | test.ps1:8:1:8:9 | pre-return value for Sink |
| test.ps1:8:1:8:9 | implicit unwrapping of Sink | test.ps1:1:1:11:8 | return value for test.ps1 |
| test.ps1:8:1:8:9 | pre-return value for Sink | test.ps1:8:1:8:9 | implicit unwrapping of Sink |
| test.ps1:10:1:10:3 | c | test.ps1:11:6:11:8 | c |
| test.ps1:10:6:10:16 | [...]... | test.ps1:10:1:10:3 | c |
| test.ps1:10:6:10:16 | [...]... | test.ps1:10:1:10:16 | ...=... |
| test.ps1:10:6:10:16 | [...]... | test.ps1:10:6:10:16 | [...]... |
| test.ps1:10:14:10:16 | b | test.ps1:10:6:10:16 | [...]... |
| test.ps1:11:1:11:8 | Sink | test.ps1:11:1:11:8 | pre-return value for Sink |
| test.ps1:11:1:11:8 | Sink | test.ps1:11:1:11:8 | pre-return value for Sink |
| test.ps1:11:1:11:8 | implicit unwrapping of Sink | test.ps1:1:1:11:8 | return value for test.ps1 |
| test.ps1:11:1:11:8 | pre-return value for Sink | test.ps1:11:1:11:8 | implicit unwrapping of Sink |
